### DISCUSSION TITLE
HID interface number

### DISCUSSION THREAD

**Discussion by ObligedTester** [hathach/tinyusb#2386] [General]
Hello,

I recently tested a couple of keyboards with my small RP2040 implementation. I noted that one of the keyboards got two HID interface IDs assigned.

Log output of when the keyboard is attached:

```
Parsing Configuration descriptor (wTotalLength = 59)
[1] HID opening Interface 0
  Open EP 81 with Size = 8
Allocate interrupt ep 0
  HID opened
  Bind EP 81 to driver id 0
[1] HID opening Interface 1
  Open EP 82 with Size = 8
Allocate interrupt ep 1
  HID opened
  Bind EP 82 to driver id 0
```
All the other keyboards just gets one interface ("0").
This is a very standard keyboard, one of the cheapest OEM lenovo there is. Just wondering what the reason is for this? 

First i was thinking it might be registering as two devices if the controller used in it might be used for mouse integration or something like that.
But plugging it in to my linux PC, only one device is registered and its only called a keyboard.

Also, any tip on the best way to differentiate the separate instances in the code? For example writing something to instance 0 (capslock LED) nothing will happen.

Any thoughts is appreciated!

**Comment by hathach** [comment]
it is normal for keyboard/mouse has several hid interfaacess, the fact, I encounter multiple interface more often than not. Each interface has an index, you can check its interface number and/or parse its hid descriptor to know what they did and handle accordingly.
